smarts/core/controllers/trajectory_tracking_controller.py

```diff
@@ -208,14 +208,43 @@ def perform_trajectory_tracking_PD(
         lateral_gain = lerp(3, final_lateral_gain, normalized_speed)
         heading_gain = lerp(0.03, final_heading_gain, normalized_speed)
```

Comment:
`lateral_gain` and `heading_gain` are being reassigned on line 225 and 226 without being used

https://github.com/huawei-noah/SMARTS/pull/1205/files#diff-4c1da4781052a940dd05e04f2ebf416b41ddf16f511b16e87fe1c41fa2dd6946R225-R226
```suggestion

```

Reply:
Ok, I did a bit more reorg to clean this up some more.  See what you think now.